### PR TITLE
Restructure core/base slightly

### PR DESCRIPTION
**Pull request by jack-w-shaw**
Extra all series related functions into series.go. Keep all offenders in 1 place!

Remove some unused functions and make some functions private

There are now only two (public) functions which parse series, GetSeriesFromBase and GetBaseFromSeries

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

Juju compiles

All unit tests pass

```
go test github.com/juju/juju/core/base
```

### CODE REVIEW

File: core/base/base.go [hunk 1]
```diff
@@ -131,46 +131,3 @@ func (b Base) DisplayString() string {
...
-}
-
-// GetSeriesFromChannel gets the series from os name and channel.
-func GetSeriesFromChannel(name string, channel string) (string, error) {
-	base, err := ParseBase(name, channel)
-	if err != nil {
-		return "", errors.Trace(err)
-	}
-	return GetSeriesFromBase(base)
-}
```
Comment (jack-w-shaw) [inline core/base/base.go:158]: :zap:

File: core/base/supportedseries.go [hunk 2]
```diff
@@ -37,36 +34,6 @@ func getAllSeriesVersions() map[SeriesName]seriesVersion {
...
-
-// DefaultOSTypeNameFromSeries returns the operating system based
-// on the given series, defaulting to Ubuntu for unknown series.
-func DefaultOSTypeNameFromSeries(series string) string {
-	osType, err := GetOSFromSeries(series)
-	if err != nil {
-		osType = ostype.Ubuntu
-	}
-	return strings.ToLower(osType.String())
-}
```
[the remark this answers is not in the document]
Comment (jack-w-shaw) [reply core/base/supportedseries.go:68]: (:zap: means this function has been removed)

**Review by cderici — APPROVED** [review]
LGTM 👍

**Pull request merged**